chips/lowrisc/src/i2c.rs

```diff
@@ -0,0 +1,275 @@
+        (0x30 => timing1: ReadWrite<u32, TIMING1::Register>),
+        (0x34 => timing2: ReadWrite<u32, TIMING2::Register>),
+        (0x38 => timing3: ReadWrite<u32, TIMING3::Register>),
+        (0x3C => timing4: ReadWrite<u32, TIMING4::Register>),
+        (0x40 => timeout_ctrl: ReadWrite<u32, TIMEOUT_CTRL::Register>),
+        (0x44 => @END),
+    }
+}
+
+register_bitfields![u32,
+    INTR [
+        FMT_WATERMARK OFFSET(0) NUMBITS(1) [],
+        RX_WATERMARK OFFSET(1) NUMBITS(1) [],
+        FMT_OVERFLOW OFFSET(2) NUMBITS(1) [],
+        RX_OVERFLOW OFFSET(3) NUMBITS(1) [],
+        NAK OFFSET(4) NUMBITS(1) [],
+        SCL_INTERFERENCE OFFSET(5) NUMBITS(1) [],
+        SDA_INTERFERENCE OFFSET(6) NUMBITS(1) [],
+        STRETCH_TIMEOUT OFFSET(7) NUMBITS(1) [],
+        SDA_UNSTABLE OFFSET(8) NUMBITS(1) []
+    ],
+    CTRL [
+        ENABLEHOST OFFSET(0) NUMBITS(1) []
+    ],
+    STATUS [
+        FMTFULL OFFSET(0) NUMBITS(1) [],
+        RXFULL OFFSET(1) NUMBITS(1) [],
+        FMTEMPTY OFFSET(2) NUMBITS(1) [],
+        HOSTIDLE OFFSET(3) NUMBITS(1) [],
+        TARGETIDLE OFFSET(4) NUMBITS(1) [],
+        RXEMPTY OFFSET(5) NUMBITS(1) []
+    ],
+    RDATA [
+        RDATA OFFSET(0) NUMBITS(8) []
+    ],
+    FDATA [
+        FBYTE OFFSET(0) NUMBITS(8) [],
+        START OFFSET(8) NUMBITS(1) [],
+        STOP OFFSET(9) NUMBITS(1) [],
+        READ OFFSET(10) NUMBITS(1) [],
+        RCONT OFFSET(11) NUMBITS(1) [],
+        NAKOK OFFSET(12) NUMBITS(1) []
+    ],
+    FIFO_CTRL [
+        RXRST OFFSET(0) NUMBITS(1) [],
+        FMTRST OFFSET(1) NUMBITS(1) [],
+        RXILVL OFFSET(2) NUMBITS(3) [
+            RXLVL1 = 0,
+            RXLVL4 = 1,
+            RXLVL8 = 2,
+            RXLVL16 = 3,
+            RXLVL30 = 4
+        ],
+        FMTILVL OFFSET(5) NUMBITS(3) [
+            FMTLVL1 = 0,
+            FMTLVL4 = 1,
+            FMTLVL8 = 2,
+            FMTLVL16 = 3,
+            FMTLVL30 = 4
+        ]
+    ],
+    FIFO_STATUS [
+        FMTLVL OFFSET(0) NUMBITS(6) [],
+        RXLVL OFFSET(16) NUMBITS(6) []
+    ],
+    OVRD [
+        TXOVRDEN OFFSET(0) NUMBITS(1) [],
+        SCLVAL OFFSET(1) NUMBITS(1) [],
+        SDAVAL OFFSET(2) NUMBITS(1) []
+    ],
+    VAL [
+        SCL_RX OFFSET(0) NUMBITS(1) [],
+        SDA_RX OFFSET(1) NUMBITS(1) []
+    ],
+    TIMING0 [
+        THIGH OFFSET(0) NUMBITS(16) [],
+        TLOW OFFSET(16) NUMBITS(16) []
+    ],
+    TIMING1 [
+        T_R OFFSET(0) NUMBITS(16) [],
+        T_F OFFSET(16) NUMBITS(16) []
+    ],
+    TIMING2 [
+        TSU_STA OFFSET(0) NUMBITS(16) [],
+        THD_STA OFFSET(16) NUMBITS(16) []
+    ],
+    TIMING3 [
+        TSU_DAT OFFSET(0) NUMBITS(16) [],
+        THD_DAT OFFSET(16) NUMBITS(16) []
+    ],
+    TIMING4 [
+        TSU_STO OFFSET(0) NUMBITS(16) [],
+        T_BUF OFFSET(16) NUMBITS(16) []
+    ],
+    TIMEOUT_CTRL [
+        VAL OFFSET(0) NUMBITS(31) [],
+        EN OFFSET(31) NUMBITS(1) []
+    ]
+];
+
+pub struct I2c<'a> {
```

Comment:
It looks like Tock writes this `i2c` everywhere else; is there a reason for different capitalization here?

Reply:
I was just following CamelCase for the struct names. Should I change it to all lower case?

Reply:
Sigh. I don't _really_ care that much. This is firmly on the list of things I want rustfmt to decide for us, I was just surprised that it was allowed to be different than elsewhere in Tock.

Leaving it camel case to follow Rust conventions is probably fine.